File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()

        for i, val in enumerate(nums):
            if val > 0:
                break
            
            if i > 0 and val == nums[i-1]:
                continue
            
            left = i + 1
            right = len(nums) - 1

            while left < right:
                threeSum = val + nums[left] + nums[right]
                if threeSum > 0:
                    right -= 1
                elif threeSum < 0:
                    left += 1
                else:
                    res.append([val,nums[left],nums[right]])
                    left += 1
                    right -= 1
                    while nums[left] == nums[left - 1] and left < right:
                        left += 1
        return res

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        std:vector<vector<int>> res;
        if(!root){
            return res;
        }
        std::deque<TreeNode*> q = {root};

        while(!q.empty()){
            int size = q.size();
            std::vector<int> ans;
            for(int i = 0; i < size; i++){
                TreeNode* node = q.front();
                ans.emplace_back(node->val);
                q.pop_front();
                if(node->left){
                    q.push_back(node->left);
                }
                if(node->right){
                    q.push_back(node->right);
                }
            }
            res.emplace_back(ans);
        }
        return res;
    }
};
